### PR TITLE
 pkgs/development: switch to fetchFromGitHub part 3

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
use `fetchFromGitHub`

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/pa_ounit/default.nix [hunk 1]
```diff
@@ -1,18 +1,18 @@
-{ lib, buildOcaml, ocaml, fetchurl, ounit }:
-
-if lib.versionAtLeast ocaml.version "4.06"
-then throw "pa_ounit is not available for OCaml ${ocaml.version}"
-else
```
Comment (Stunkymonkey) [inline pkgs/development/ocaml-modules/pa_ounit/default.nix:5]: i do not know for sure if added line 14 provides the same

Comment (dali99) [reply pkgs/development/ocaml-modules/pa_ounit/default.nix:5]: It does :smile:

Comment (Stunkymonkey) [reply pkgs/development/ocaml-modules/pa_ounit/default.nix:5]: or should it be `minimumSupportedOcamlVersion`? i am confused with the different namings that exist here.

Comment (dali99) [reply pkgs/development/ocaml-modules/pa_ounit/default.nix:5]: oh woops you're right. in `buildOcaml` it's `minimumSupportedOcamlVersion`
In `buildDunePackage` and `buildOasisPackage` it's `minimumOCamlVersion`

Comment (Stunkymonkey) [reply pkgs/development/ocaml-modules/pa_ounit/default.nix:5]: is it desired to change this to a single variable? because all do the exact same thing.

Comment (dali99) [reply pkgs/development/ocaml-modules/pa_ounit/default.nix:5]: I wouldnt be opposed to it

Comment (Stunkymonkey) [reply pkgs/development/ocaml-modules/pa_ounit/default.nix:5]: I just looked over it again. I think the old code does the exact opposite like a `maximumOCamlVersion`. so i will leave the old code and address this issue in a new PR.

Comment (dali99) [reply pkgs/development/ocaml-modules/pa_ounit/default.nix]: :facepalm: you're quite right

Comment (Stunkymonkey) [reply pkgs/development/ocaml-modules/pa_ounit/default.nix]: anything to add for review?

**Review by D4ndellion — APPROVED** [review]
5 packages built:
ocamlPackages.eliom ocamlPackages.js_of_ocaml-tyxml ocamlPackages.ocsigen-start ocamlPackages.ocsigen-toolkit ocamlPackages.reactivedata

Also tested building the some of the packages not picked up by nix-review.

File: pkgs/development/ocaml-modules/pa_bench/default.nix [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 
   minimumSupportedOcamlVersion = "4.00";
 
-  src = fetchurl {
-    url = "https://github.com/janestreet/pa_bench/archive/${version}.tar.gz";
-    sha256 = "1cd6291gdnk6h8ziclg6x3if8z5xy67nfz9gx8sx4k2cwv0j29k5";
+  src = fetchFromGitHub {
+    owner = "janestreet";
+    repo = "pa_bench";
+    rev = "113.00.00";
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/pa_bench/default.nix:12]:
```suggestion
    rev = version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 143870` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>5 packages failed to build and are new build failures:</summary>
  <ul>
<li>ocamlPackages.eliom: log was empty</li>
<li>ocamlPackages.js_of_ocaml-tyxml: log was empty</li>
<li>ocamlPackages.ocsigen-start: log was empty</li>
<li>ocamlPackages.ocsigen-toolkit: log was empty</li>
<li>ocamlPackages.reactivedata: log was empty</li>
  </ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
ocamlPackages.eliom:
<details>


warning: unclear-gpl
`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/ocaml-modules/eliom/default.nix:54:5:

```
   |
54 |     license = lib.licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: build-tools-in-build-inputs
which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/ocaml-modules/eliom/default.nix:21:3:

```
   |
21 |   buildInputs = [ ocaml which findlib js_of_ocaml-ocamlbuild
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
</details>
ocamlPackages.js_of_ocaml-tyxml:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/development/tools/ocaml/js_of_ocaml/compiler.nix:24:5:

```
   |
24 |     license = lib.licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
warning: unused-argument
Unused argument: `ocaml`.
Near pkgs/development/tools/ocaml/js_of_ocaml/compiler.nix:2:3:

```
  |
2 | , ocaml, cmdliner, cppo, yojson, ppxlib
  |   ^
```
</details>
ocamlPackages.ocsigen-start:
<details>


warning: name-and-version
Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/development/ocaml-modules/ocsigen-start/default.nix:8:3:

```
  |
8 |   name = "ocaml${ocaml.version}-ocsigen-start-${version}";
  |   ^
```

Near pkgs/development/ocaml-modules/ocsigen-start/default.nix:9:3:

```
  |
9 |   version = "4.3.0";
  |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md
</details>
ocamlPackages.ocsigen-toolkit:
<details>


warning: unclear-gpl
`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/ocaml-modules/ocsigen-toolkit/default.nix:32:5:

```
   |
32 |     license = lib.licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
ocamlPackages.reactivedata:
<details>


warning: unclear-gpl
`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/ocaml-modules/reactivedata/default.nix:28:5:

```
   |
28 |     license = licenses.lgpl21;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

File: pkgs/development/ocaml-modules/reactivedata/default.nix [hunk 3]
```diff
@@ -8,9 +8,11 @@ stdenv.mkDerivation rec {
...
   version = "0.2.2";
 
-  src = fetchurl {
-    url = "https://github.com/ocsigen/reactiveData/archive/${version}.tar.gz";
-    sha256 = "0jzagyp4zla28wykvcgqwd8df71ir0vb4s8akp02cfacd5v86sng";
+  src = fetchFromGitHub {
+    owner = "ocsigen";
+    repo = "reactiveData";
+    rev = version;
+    sha256 = "sha256-z2qDdmlMOSbAnQppsjbIMRzXUOP4sT09R0LRT65/6ks=";
```
Comment (SuperSandro2000) [inline pkgs/development/ocaml-modules/reactivedata/default.nix:15]:
```suggestion
    sha256 = "sha256-YLkacIbjxZQ/ThgSxjTqviBYih6eW2GX5H7iybQDv1A=";
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 143870` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>ocamlPackages.eliom</li>
<li>ocamlPackages.js_of_ocaml-tyxml</li>
<li>ocamlPackages.ocsigen-start</li>
<li>ocamlPackages.ocsigen-toolkit</li>
<li>ocamlPackages.reactivedata</li>
</ul>
</details>

**Pull request merged**